composer/datasets/ade20k.py

```diff
@@ -0,0 +1,300 @@
+            image = TF.adjust_contrast(image, contrast_factor)  # type: ignore
+
+        return image
+
+
+class ADE20k(Dataset):
+    """PyTorch Dataset for ADE20k.
+
+    Args:
+        datadir (str): the path to the ADE20k folder.
+        split (str): the dataset split to use, either 'train', 'val', or 'test'. Default is 'train'.
+        both_transforms (torch.nn.Module): transformations to apply to the image and target simultaneously.
+            Default is None.
+        image_transforms (torch.nn.Module): transformations to apply to the image only. Default is None.
+        target_transforms (torch.nn.Module): transformations to apply to the target only. Default is None.
+    """
+
+    def __init__(self,
+                 datadir: str,
+                 split: str = 'train',
+                 both_transforms: torch.nn.Module = None,
+                 image_transforms: torch.nn.Module = None,
+                 target_transforms: torch.nn.Module = None):
+        super().__init__()
+        self.datadir = datadir
+        self.split = split
+        self.both_transforms = both_transforms
+        self.image_transforms = image_transforms
+        self.target_transforms = target_transforms
+
+        # Check datadir value
+        if self.datadir is None:
+            raise ValueError("datadir must be specified")
+        elif not os.path.exists(self.datadir):
+            raise FileNotFoundError(f"datadir path does not exist: {self.datadir}")
+
+        # Check split value
+        if self.split not in ["train", "val", "test"]:
+            raise ValueError(f"split must be one of [`train`, `val`, `test`] but is: {self.split}")
+
+        self.image_dir = os.path.join(self.datadir, 'images', self.split)
+        if not os.path.exists(self.image_dir):
+            raise FileNotFoundError(f"ADE20k directory structure is not as expected: {self.image_dir} does not exist")
+
+        self.image_files = os.listdir(self.image_dir)
+
+        # Filter for ADE files
+        self.image_files = [f for f in self.image_files if f[:3] == 'ADE']
+
+        # Remove grayscale samples
+        if self.split == 'train':
+            corrupted_samples = ['00003020', '00001701', '00013508', '00008455']
+            for sample in corrupted_samples:
+                sample_file = f'ADE_train_{sample}.jpg'
+                if sample_file in self.image_files:
+                    self.image_files.remove(sample_file)
+
+    def __getitem__(self, index):
+        # Load image
+        image_file = self.image_files[index]
+        image_path = os.path.join(self.image_dir, image_file)
+        image = Image.open(image_path)
+
+        # Load annotation target if using either train or val splits
+        if self.split in ['train', 'val']:
+            target_path = os.path.join(self.datadir, 'annotations', self.split, image_file.split('.')[0] + '.png')
+            target = Image.open(target_path)
+
+            if self.both_transforms:
```

Comment:
It isn't necessary, but it felt weird for the photometric distortions to take a target argument since the target would likely never be used during the transformation. Also, the `PadToSize` function would need to take a separate `fill` arguments for the image and target which feels a bit weird. The changes would decrease downstream code for this specific use case, but would add unnecessary attributes to the transformations.

How strongly do you feel about simplifying the transformation composition? Happy to change the transformations if you think it makes more sense.

Reply:
I do prefer it to changing the dataset interface but I can see the benefits on being more explicit about exactly which transformation is applied. I think this is ok for now but let's think about it more as we add more datasets that need to transform both the input and targets.